libraries/AP_NavEKF3/AP_NavEKF3_OptFlowFusion.cpp

```diff
@@ -52,7 +52,7 @@ void NavEKF3_core::SelectFlowFusion()
 
     // Fuse optical flow data into the main filter
     if (flowDataToFuse && tiltOK) {
-        if (frontend->_flowUse == FLOW_USE_NAV) {
+        if ((frontend->_flowUse == FLOW_USE_NAV) && (frontend->_sources.getVelXYSource() == AP_NavEKF_Source::SourceXY::OPTFLOW)) {
```

Comment:
Yes, although the user is required to enable the use of optical flow via the parameter, it can be inhibited via the source selection interface.

Reply:
yes, previously enabling a sensor would likely mean it would be fused.  Now the user has to specify that it should be fused.  The advantage of this is that it gives us the power to decide when it should be fused.  So for example if we know ExtNav is good the pilot (or lua script) can decide not to fuse optical flow.